### PR TITLE
Studio: Fix offline mode on push and pull buttons

### PR DESCRIPTION
**Pull request by katinthehatsite**
## Related issues

Closes https://github.com/Automattic/dotcom-forge/issues/9939

## Proposed Changes

This PR adds back the offline mode to `Pull` and `Push` buttons (it was initially added in https://github.com/Automattic/dotcom-forge/issues/9808 but unintentionally removed in https://github.com/Automattic/dotcom-forge/issues/9939 ).

Screenshot 2024-11-25 at 12 03 44 PM

## Testing Instructions

* Pull the changes from this branch
* Start Studio app with `STUDIO_SITE_SYNC=true npm start`
* Navigate to the Sync tab that has some sites connected
* Disable your internet connection (in my case, it was enough to disconnect from WiFi)
* Confirm that when you hover over `Pull` or `Push` buttons, they are disabled and you can see a tooltip with information

## Pre-merge Checklist

- [ ] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: src/components/sync-connected-sites.tsx [hunk 1]
```diff
@@ -248,37 +250,44 @@ const SyncConnectedSitesSection = ( {
...
-												disabled={ isAnySitePulling || isAnySitePushing }
-											>
-												<Icon icon={ cloudUpload } />
-												{ __( 'Push' ) }
-											</Button>
-										</div>
+										<Tooltip
+											disabled={ ! isOffline }
+											icon={ offlineIcon }
+											text={ __( 'Pulling or pushing a site requires an internet connection.' ) }
```
Comment (ivan-ottinger) [inline src/components/sync-connected-sites.tsx:266]: I have noticed that sometimes we add period (`.`) at the end of tooltip strings, sometimes we omit it.

For instance, in the recent PR (https://github.com/Automattic/studio/pull/684) with the last push / pull time, we don't add the dot.

**Review by ivan-ottinger — APPROVED** [review]
The proposed changes look great and add tooltip with correct styling on the Pull / Push buttons when offline. 👌🏼🙂 

Sharing just one observation below.

Update: There's also a conflict now after the recent PR merge.

**Review by nightnei — APPROVED** [review]
LGTM, thanks for addressing it 👍

**Review by nightnei — APPROVED** [review]

File: src/components/sync-connected-sites.tsx (hunk 1 above)
Comment (katinthehatsite) [reply src/components/sync-connected-sites.tsx:266]: Yeah I noticed that one as well. Let's remove the `.` to ensure that we are consistent with other tooltips such as `Open settings` etc.

Comment (katinthehatsite) [reply src/components/sync-connected-sites.tsx:266]: Actually, let's keep the period and add it to the other PR. I checked UI around the app and it seems that we are using it in the sentence formulation e.g. `You're currently online.` vs. not using it for things like `Open settings`. I think it makes sense to keep this pattern.

Comment (ivan-ottinger) [reply src/components/sync-connected-sites.tsx:266]: That sounds good to me. 👍🏼

**Pull request merged**